src/coreclr/vm/jitinterface.cpp

```diff
@@ -5945,6 +5987,40 @@ CorInfoHelpFunc CEEInfo::getUnBoxHelper(CORINFO_CLASS_HANDLE clsHnd)
     return CORINFO_HELP_UNBOX;
 }
 
+/***********************************************************************/
+void* CEEInfo::getRuntimeTypePointer(CORINFO_CLASS_HANDLE clsHnd)
+{
+    LIMITED_METHOD_CONTRACT;
+
+    void* pointer = nullptr;
+
+    JIT_TO_EE_TRANSITION();
+
+    GCX_COOP();
+
+    TypeHandle typeHnd(clsHnd);
+    if (!typeHnd.IsNull() && !typeHnd.IsTypeDesc() && typeHnd.IsFullyLoaded())
```

Comment:
You should be able to do this optimization for TypeDescs too.

Also, IsFullyLoaded check should not be needed. Types handed back to the JIT are always fully loaded.